File: apis/csharp/routine-api/exception/RoutineException.cs
using System;

namespace routine_api.exception
{
    /// <summary>
    /// Exception thrown when there is an error executing a routine.
    /// </summary>
    public class RoutineException : Exception
    {
        public RoutineException(string message) : base(message)
        {
        }
        
        public RoutineException(Exception cause) : base(cause.Message, cause)
        {
        }

        public RoutineException(string message, Exception cause) : base(message, cause)
        {
        }
    }
}

File: routine-api/src/main/java/at/enfilo/def/routine/RoutineException.java
package at.enfilo.def.routine;

public class RoutineException extends RuntimeException {
	public RoutineException() {
	}

	public RoutineException(String message) {
		super(message);
	}

	public RoutineException(String message, Throwable cause) {
		super(message, cause);
	}

	public RoutineException(Throwable cause) {
		super(cause);
	}
}
